test/streamingReceiver.spec.ts

```diff
@@ -657,137 +496,109 @@ describe("Streaming Receiver from Queue/Subscription", function(): void {
+    await testPeekMsgsLength(deadletterClient, 0);
+  }
+
+  it("Queue: deadLetter() moves message to deadletter queue", async function(): Promise<void> {
+    await testDeadletter(queueClient, queueClient, deadletterQueueClient, false);
   });
 
-  it("With auto-complete disabled, dead lettering the message results in not getting the same message again from subscription. The message is then gotten only from the dead letter", async function(): Promise<
+  it("Subscription: deadLetter() moves message to deadletter queue", async function(): Promise<
     void
   > {
-    await topicClient.sendBatch(testMessages);
-
-    const receiveListener = await subscriptionClient.receive(
-      (msg: ServiceBusMessage) => {
-        return msg.deadLetter();
-      },
-      (err: Error) => {
-        should.not.exist(err);
-      },
-      { autoComplete: false }
-    );
-
-    await delay(4000);
-
-    await receiveListener.stop();
-
-    await testPeekMsgsLength(subscriptionClient, 0);
-    const deadLetterSubscriptionPath = Namespace.getDeadLetterSubcriptionPathForSubcription(
-      topicClient.name,
-      subscriptionClient.subscriptionName
-    );
-
-    const deadletterSubscriptionClient = namespace.createSubscriptionClient(
-      deadLetterSubscriptionPath ? deadLetterSubscriptionPath : "",
-      subscriptionClient.subscriptionName
-    );
+    await testDeadletter(topicClient, subscriptionClient, deadletterSubscriptionClient, false);
+  });
 
-    await testPeekMsgsLength(deadletterSubscriptionClient, 2); // Two messages in the DL
+  it("Queue with autoComplete: deadLetter() moves message to deadletter queue", async function(): Promise<
+    void
+  > {
+    await testDeadletter(queueClient, queueClient, deadletterQueueClient, false);
+  });
 
-    const deadLetterMsgs = await deadletterSubscriptionClient.receiveBatch(2);
-    should.equal(Array.isArray(deadLetterMsgs), true);
-    should.equal(deadLetterMsgs.length, 2);
-    should.equal(deadLetterMsgs[0].messageId, testMessages[0].messageId);
-    should.equal(deadLetterMsgs[1].messageId, testMessages[1].messageId);
+  it("Subscription with autoComplete: deadLetter() moves message to deadletter queue", async function(): Promise<
+    void
+  > {
+    await testDeadletter(topicClient, subscriptionClient, deadletterSubscriptionClient, false);
+  });
+});
 
-    await deadLetterMsgs[0].complete();
-    await deadLetterMsgs[1].complete();
+describe("Multiple Streaming Receivers", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
 
-    await testPeekMsgsLength(deadletterSubscriptionClient, 0);
+  afterEach(async () => {
+    await afterEachTest();
   });
 
-  it("Second Streaming Receiver call should fail if the first one is not stopped for Queues", async function(): Promise<
-    void
-  > {
-    const receiveListener: ReceiveHandler = await queueClient.receive(
+  async function testMultipleReceiveBatchCalls(
```

Comment:
The name `testMultipleReceiveBatchCalls` sounds like it belongs to the batch receiver.
Should we change that to `testMultipleReceiveCalls` or something ?